test/unit/puppetlabs/puppetserver/shell_utils_test.clj

```diff
@@ -0,0 +1,36 @@
+(ns puppetlabs.puppetserver.shell-utils-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.puppetserver.shell-utils :as sh-utils]
+            [me.raynes.fs :as fs]))
+
+(def test-resources (fs/absolute-path "./dev-resources/puppetlabs/puppetserver/shell_utils_test"))
+
+(defn script-path
+  [script-name]
+  (str test-resources "/" script-name))
+
+(deftest returns-the-exit-code
+  (testing "true should return 0"
+    (is (zero? (:exit-code (sh-utils/execute-command (script-path "true"))))))
+  (testing "false should return 1"
+    (is (= 1 (:exit-code (sh-utils/execute-command (script-path "false")))))))
+
+(deftest returns-stdout-correctly
+  (testing "echo should add content to stdout"
+    (is (= "foo\n" (:stdout (sh-utils/execute-command (script-path "echo") ["foo"]))))))
+
+(deftest returns-stderr-correctly
+  (testing "echo can add content to stderr as well"
+    (is (= "bar\n" (:stderr (sh-utils/execute-command (script-path "warn") ["bar"]))))))
+
+(deftest pass-args-correctly
+  (testing "passes the expected number of args to cmd"
+    (is (= 5 (:exit-code (sh-utils/execute-command (script-path "num-args") ["a" "b" "c" "d" "e"]))))))
+
+(deftest throws-exception-for-non-absolute-path
+  (testing "Commands must be given using absolute paths"
+    (is (thrown? IllegalArgumentException (sh-utils/execute-command "echo")))))
+
+(deftest throws-exception-for-non-existent-file
+  (testing "The given command must exist"
+    (is (thrown? IllegalArgumentException (sh-utils/execute-command "/usr/bin/footest")))))
```

Comment:
`man pipe` on my linux box shows the default buffer size being 65536.  Doubling that for a test would only use 128k of memory, which seems not-entirely-crazypants.